src/shared/components/palette/case-link/components/read-linked-cases.component.ts

```diff
@@ -18,51 +18,71 @@ export class ReadLinkedCasesComponent implements OnInit, AfterViewInit {
-        },
-        error: error => this.getFailureNotification(error)
-      });
-    }
+  public ngOnInit(): void {
+    const reasonCodeAPIurl = this.appConfig.getRDCommonDataApiUrl() + '/lov/categories/CaseLinkingReasonCode';
+    this.commonDataService.getRefData(reasonCodeAPIurl).subscribe({
+      next: reasons => {
+        this.reasonListLoaded = true;
+        this.linkedCasesService.linkCaseReasons = reasons.list_of_values.sort((a, b) => (a.value_en > b.value_en) ? 1 : -1);
+      },
+      error: error => this.getFailureNotification(error)
+    });
+  }
 
-    public ngAfterViewInit(): void {
-      this.linkedCasesService.caseFieldValue = this.caseField.value || [];
-      let labelField = document.getElementsByClassName('govuk-heading-l');
-      if (labelField && labelField.length) {
-        labelField[0].replaceWith('')
-      }
-      labelField = document.getElementsByClassName('heading-h2');
-      if (labelField && labelField.length) {
-        labelField[0].replaceWith('')
-      }
-      labelField = document.getElementsByClassName('case-viewer-label');
-      if (labelField && labelField.length) {
-        labelField[0].replaceWith('')
-      }
+  public ngAfterViewInit(): void {
+    this.linkedCasesService.caseFieldValue = this.caseField.value || [];
+    let labelField = document.getElementsByClassName('govuk-heading-l');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+    labelField = document.getElementsByClassName('heading-h2');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
     }
+    labelField = document.getElementsByClassName('case-viewer-label');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+  }
 
   public reloadCurrentRoute() {
     let currentUrl = this.router.url;
-    this.router.navigateByUrl('/', {skipLocationChange: true}).then(() => {
-        this.router.navigate([currentUrl]);
+    this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
+      this.router.navigate([currentUrl]);
     });
   }
 
-  public getFailureNotification(evt) {
-    const errorMessage = 'There has been a system error and your request could not be processed.';
-    this.serverError = {
+  public getFailureLinkedToNotification(evt) {
+    const errorMessage = 'Some case information is not available at the moment';
+    this.serverLinkedToError = {
       id: 'backendError', message: errorMessage
     };
+    this.getFailureNotification(evt);
+  }
+
+  public getFailureLinkedFromNotification(evt) {
+    const errorMessage = 'Some case information is not available at the moment';
+    this.serverLinkedFromError = {
+      id: 'backendError', message: errorMessage
+    };
+    this.getFailureNotification(evt);
+  }
+
+  public getFailureNotification(evt) {
+    if (this.serverLinkedToError && this.serverLinkedFromError) {
```

Comment:
Is this or(||) rather than && when seeing either these two messages it is the backend error?

Reply:
Updated BE error handling